File: assignment.py
import numpy as np
import matplotlib.pyplot as plt

# Prompting the user to input the lengths of the signals
lx = int(input('Enter the length of signal x(n): '))
ly = int(input('Enter the length of signal y(n): '))

# Generating random signals x(n) and y(n)
x = np.random.rand(lx)
y = np.random.rand(ly)

# Function to compute the summation
def compute_summation(x, y, ln):
    result = []
    for k in range(0, ln, 2):
        summation = np.sum(x[:min(len(x), ln+1-k)] * y[k:min(len(y), k+ln+1)])
        result.append(summation)
    return result

# Compute the summation
ln = min(lx, ly)
result = compute_summation(x, y, ln)

# Plotting
plt.plot(result)
plt.xlabel('Index (k values)')
plt.ylabel('Summation')
plt.title('Summation from n=0 to ln of x(n)y(n+k)')
plt.show()

File: assignment.m
% Prompting the user to input the lengths of the signals
lx = input('Enter the length of signal x(n): ');
ly = input('Enter the length of signal y(n): ');
% Generating random signals x(n) and y(n)
x = rand(1, lx);
y = rand(1, ly);
% Function to compute the summation
function result = compute_summation(x, y, ln)
    result = [];
    for k = 0:2:ln
        summation = sum(x(1:min(length(x), ln+1-k)) .* y(k+1:min(length(y), k+ln+1)));
        result = [result, summation];
    end
end
% Compute the summation
ln = min(lx, ly);
result = compute_summation(x, y, ln);
% Plotting
plot(result);
xlabel('Index (k values)');
ylabel('Summation');
title('Summation from n=0 to ln of x(n)y(n+k)');